sklearn/isotonic.py

```diff
@@ -228,7 +228,7 @@ def _build_f(self, X, y):
                              .format(self.out_of_bounds))
 
         bounds_error = self.out_of_bounds == "raise"
-        self.f_ = interpolate.interp1d(X, y, kind='linear',
+        self.f_ = interpolate.interp1d(X, y, kind='slinear',
```

Comment:
you can (and should) now also undo the workaround in line 321, i.e., adding np.finfo(float).resolution to self.X_min_, and remove the comment in 319-320